File: java/algorithms/easy/birthdayCakeCandles.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-06-13
 * Last modified: 2026-06-13
 * @description A Very Big Sum - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/a-very-big-sum/problem}
 */
//---------- Imports ----------
import java.util.List;


//---------- solution ----------
class Result 
{
    /**
     * Counts how many candles are tallest.
     * 
     * @param {List<Integer>} candles
     * @return {int}
     */
    public static int birthdayCakeCandles(List<Integer> candles) 
    { 
        int count = 1;
        candles.sort((a, b) -> b - a);
        int lastNumber = candles.get(0);        

        for (int i = 1; i < candles.size(); i++)
        {
            if (lastNumber == candles.get(i))
            { count++; }
            else
            { break; }
        }//end for

        return(count);
    }//end birthdayCakeCandles()
}//end class

File: typescript-javascript/algorithms/easy/birthdayCakeCandles.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-06-15
 * Last modified: 2026-06-15
 * @description Birthday Cake Candles - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/birthday-cake-candles/problem}
 */
//---------- functions ----------
/**
 * Count how many candles are tallest in the array of candles. The tallest candles are the ones that can be blown out.
 * 
 * @param {number[]} candles 
 * @returns {number}
 */
function birthdayCakeCandles(candles: number[]): number 
{
    let count: number = 1;
    candles.sort((a, b) => b - a);
    let lastNumber: number = candles[0];
    
    for(let i = 1; i < candles.length; i++)
    {
        if(lastNumber === candles[i])
        { count++; }
        else
        { break; }   
    }
    
    return(count);
}//end birthdayCakeCandles()